src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1179,6 +1180,24 @@ private function addNewInstance(Definition $definition, string $return = '', str
                     throw new RuntimeException(sprintf('Cannot dump definition because of invalid factory method (%s).', $callable[1] ?: 'n/a'));
                 }
 
+                if (['...'] === $arguments && $definition->isLazy() && 'Closure' === ($definition->getClass() ?? 'Closure') && (
+                    $callable[0] instanceof Reference
+                    || ($callable[0] instanceof Definition && !$this->definitionVariables->contains($callable[0]))
+                )) {
+                    $this->addContainerRef = true;
+                    $class = ($callable[0] instanceof Reference ? $this->container->findDefinition($callable[0]) : $callable[0])->getClass();
+                    $r = $this->container->getReflectionClass($class);
+
+                    if ($r && $r->hasMethod($callable[1])) {
+                        $signature = ProxyHelper::exportSignature($r->getMethod($callable[1]));
```

Comment:
Does this preserve parameter attributes ?

Reply:
It doesn't, and it shouldn't: proxies don't have to repeat the attributes.

Reply:
This means that such lazy callables won't be suitable for places doing runtime usages of attributes, like controller argument resolvers.

Reply:
Correct. That's also the case for other lazy services: generated proxies don't copy attributes. Should be documented.

Reply:
Actually, it depends on the kind on proxies being generated. For lazy ghosts, we don't override methods, so they have the attributes in their inherited signatures.